Add support for Netlify deployments

hectcastro opened the pull request:
This change set includes a variety of different changes to prepare the SPA for deployment via Netlify.

- [x] Connect Netlify URL to `population-map.districtbuilder.org` domain
- [x] Reset the main deployment branch in the Netlify configuration to `master`

Fixes https://github.com/azavea/operations/issues/466

---

**Testing**

- Ensure `npm run build` output contains no linter warnings
- Ensure the [application](https://deploy-preview-17--db-population-map.netlify.app/) functions as expected when served via Netlify

hectcastro committed:
Remove yarn.lock file

There is a package-lock.json file in the repository.

hectcastro committed:
Add Netlify configuration file

hectcastro committed:
Update page title

Removes the CRA default.

hectcastro committed:
Add .nvmrc

Makes sure we consistently use Node.js v12.18.0

jfrankl commented:
Sorry, I just merged a commit that caused these two conflicts. I can address these later tonight, or you're welcome to do it if you want. My version of `netlify.toml` can be totally replaced by yours, and I'm hoping the `MapboxGLMap.js` isn't significant.

hectcastro committed:
Format with Prettier and lint with CRA

hectcastro commented on package.json:
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "underover-districts",
+  "name": "undercover-districts",
```
Not sure, but I thought this was a typo.

jfrankl replied:
Ha, it wasn't a typo: it was `underover` as in **Under**population, **over**population. But I see why you thought it was.

I will rename this to match the repo name: `us-cd-population-map`. I think that makes more sense. `underover` was a placeholder from early in the project.

hectcastro commented on src/components/MapboxGLMap.js:
```diff
@@ -435,7 +453,8 @@ const MapboxGLMap = () => {
               Use this map to see which districts are over-or-under populated.
-              For more information, read the corresponding <a href="#">blog</a>.
+              For more information, read the corresponding
+              <a href="https://www.azavea.com/blog/">blog</a>.
```
I believe that this line needs to be updated once the blog post is published.

hectcastro reviewed (commented):
Merge conflicts should be gone now—I rebased this branch with your changes on `master`.

Commit:
Fix project name in package.json

Switch name to match the repo name. `underover` was a holdover from an earlier version of the app.

Commit:
Add missing space

The space between this word and link was being removed by `prettier` or a different script.

jfrankl reviewed (approved):
Works great. Thank you for making this happen on short notice.

hectcastro commented:
Np. The site should be up on https://population-map.districtbuilder.org/ now.
